test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -1039,9 +1039,11 @@ private Object runExtensionMethod(ReflectiveInvocationContext<Method> invocation
                         cloneRequired = false;
                     } else if (TestInfo.class.isAssignableFrom(theclass)) {
                         TestInfo info = (TestInfo) arg;
-                        Method newTestMethod = determineTCCLExtensionMethod(info.getTestMethod().get(), testClassFromTCCL);
+                        Method newTestMethod = info.getTestMethod().isPresent()
+                                ? determineTCCLExtensionMethod(info.getTestMethod().get(), testClassFromTCCL)
```

Comment:
Note: I'm usually a fan of `Optional.map()` but `determineTCCLExtensionMethod()` throws a checked exception and I didn't want to change that.